R/step_adjust_latency.R

```diff
@@ -13,30 +13,42 @@
 #'   predictors and outcome.
 #' @param method a character. Determines the method by which the
-#'   forecast handles latency. All of these assume the forecast date is the
-#'   `as_of` of the `epi_df`. The options are:
+#'   forecast handles latency. The options are:
 #'   - `"extend_ahead"`: Lengthen the ahead so that forecasting from the last
-#'   observation results in a forecast `ahead` after the `as_of` date. E.g. if
-#'   there are 3 days of latency between the last observation and the `as_of`
-#'   date for a 4 day ahead forecast, the ahead used in practice is actually 7.
+#'   observation results in a forecast `ahead` after the `forecast_date` date.
+#'   E.g. if there are 3 days of latency between the last observation and the
+#'   `forecast_date` date for a 4 day ahead forecast, the ahead used in practice
+#'   is actually 7.
 #'   - `"locf"`: carries forward the last observed value(s) up to the forecast
 #'   date. See the Vignette TODO for equivalents using other steps and more
 #'   sophisticated methods of extrapolation.
 #'   - `"extend_lags"`: per `epi_key` and `predictor`, adjusts the lag so that
 #'   the shortest lag at predict time is at the last observation. E.g. if the
 #'   lags are `c(0,7,14)` for data that is 3 days latent, the actual lags used
 #'   become `c(3,10,17)`.
+#' @param epi_keys_checked a character vector. A list of keys to group by before
+#'   finding the `max_time_value`; only used if both `fixed_latency` and
+#'   `fixed_forecast_date` are `NULL`.  The default value of this is
+#'   `c("geo_value")`, but it can be any collection of `epi_keys`.  Different
+#'   locations may have different latencies; to produce a forecast at every
+#'   location, we need to use the largest latency across every location; this
+#'   means taking `max_time_value` to be the minimum of the `max_time_value`s
+#'   for each `geo_value` (or whichever collection of keys are specified).  If
+#'   `NULL` or an empty character vector, it will take the maximum across all
+#'   values, irrespective of any keys.
 #' @param fixed_latency either a positive integer, or a labeled positive integer
-#'   vector. Cannot be set at the same time as `fixed_asof`.   If non-`NULL`,
-#'   the amount to offset the ahead or lag by. If a single integer, this is used
-#'   for all columns; if a labeled vector, the labels must correspond to the
-#'   base column names.  If `NULL`, the latency is the distance between the
-#'   `epi_df`'s `max_time_value` and either the `fixed_asof` or the `epi_df`'s
-#'   `as_of` field.
-#' @param fixed_asof either a date of the same kind used in the `epi_df`, or
-#'   NULL. Cannot be set at the same time as `fixed_latency`. If a date, it
-#'   gives the date from which the forecast is actually occurring. If `NULL`,
-#'   the `as_of` is determined either from `fixed_latency` or automatically.
+#'   vector. Cannot be set at the same time as `fixed_asof`. If non-`NULL`, the
```

Comment:
wasn't `fixed_asof` removed?

Reply:
renamed to `fixed_forecast_date`, good catch thanks